### ISSUE TITLE
Launch Rstudio add-in gives "Error: RStudio not running"

### ISSUE THREAD

**Issue by albert-ying** [Ikuyadeu/vscode-R#605]
Hi, I did what is suggested in the document to activate Rstudio addin support.

After I call the command `R: launch Rstudio addin`, I can see a list of addins in the command palette.

<img width="858" alt="image" src="https://user-images.githubusercontent.com/59846322/113540421-1e4ad180-95ae-11eb-8dcd-d2bcb0d92437.png">

But after I select it, it gives me an error in the terminal

```
[ins] r$> xaringan:::inf_mr()                                                                             
Error: RStudio not running
```

I also tried other addins listed, but they give me the same error...

Did I miss anything here?

Thank you!

**Comment by MilesMcBain** [comment]
This addin works for me.

The error suggests the `rstudioapi` functions have not been loaded by VSCode, however the fact that you are able to get to the addin menu suggests they have.

I can reproduce your behaviour by doing the following:

1. Start an R terminal with the VSCode extension using either command `R: Attach active terminal` or `R: Create new terminal`.
2. Create a new default terminal (running bash or powershell) and run the R executable with the command `R` or `R.exe`
3. With the terminal from step 2. active, run the addin from the menu and get your error.

The addin picker just passes a command to be run in the active terminal. If that active terminal is not one that has been initialised (attached) by the VSCode-R extension, RStudio addins, and many other VSCode-R extension features, will not work as expected.

Could running an R session not initialised by the vscode-R extension be a contributing factor in your case?

**Comment by albert-ying** [comment]
Thank you @MilesMcBain 

I tried again, and it was not the scenario you described, as esquisser works

```
[ins] r$> .vsc.attach()                                                                                      

[ins] r$> xaringan:::inf_mr()                                                                                
Error: RStudio not running

[ins] r$> esquisse:::esquisser()                                                                             
Loading required package: shiny

Listening on http://127.0.0.1:5829
Browsing http://127.0.0.1:5829
```

Could it be something related to my .Rprofile or VS-code setting? Also, you mentioned this addin works for you, did you use the newest version? As I just installed it yesterday, it might be something about it?
```
remotes::install_github('yihui/xaringan')
```

Thank you so much!

**Comment by MilesMcBain** [comment]
If xaringan were doing something weird in the latest version I would expect a different error about rstudioapi functionality not supported by vscode.

I'm thinking it could be your .Rprofile. I've seen situations where things in the profile cause the attach process to terminate abnormally.

Do you mind sharing what you have in there?

**Comment by albert-ying** [comment]
Ah-ha, I figured it out, but have no idea why.

So I played with my .Rprofile for a while, and found out that it was due to `library(tidyverse)`.

After I comment out this single line, the error disappeared.

**Comment by MilesMcBain** [comment]
Oh yes okay this makes sense to me. On certain platforms `tidyverse` makes a call to see if the `rstudioapi` is available and what version it is for the purposes of working out how many terminal colours it can use and if icons are supported. 

The `rstudioapi` emulation works by injecting functions into the package namespace as part of a custom 'on load' hook. So it seems that because the `tidyverse` already loaded the `rstudioapi` namespace, our custom hook doesn't get a chance to fire.

We might be able to cater for this by running the hook manually if we detect the namespace is already loaded.

**Comment by MilesMcBain** [comment]
fixed in #610
